### ISSUE TITLE
Messages validator is too strict excluding dot form keys

### ISSUE THREAD

**Issue by Eccenux** [mozilla/addons-linter#1918]
### Describe the problem and steps to reproduce it:

Extension with messages like:
```
	"options.Parameter.Name": {
		"message": "Nazwa"
	},
```

### What happened?
The linter is complaining about the message.

### What did you expect to happen?
It was fine last time I submitted my extension. It works fine in Firefox. It validates as JSON. You can find similar keys in e.g. VSCode settings... I see no reason why "." would be an invalid character.

**Comment by Eccenux** [comment]
BTW. If you are really committed to this change then I think it would be a good idea to first deprecated something and make a warning in the linter. And only after few months (or years) make it an error.

I can't publish my extensions because of this check and I think Angular update is more important then having keys with dots or not.

**Comment by AlexandraMoga** [comment]
@EnTeQuAk  I've tried to verify this with an extension containing messages like `"options.Parameter.Name"` in the `_locales` file but I am still receiving an error:

![image](https://user-images.githubusercontent.com/31961530/37700799-8ba149d2-2cf5-11e8-8ab9-e0a6d0216d7c.png)


Here is the full validation report: https://addons.allizom.org/en-US/developers/upload/e15cd1ba35c341779df38926b234db7e

**Comment by EnTeQuAk** [comment]
@AlexandraMoga this only landed a few minutes ago on -dev probably and will land on -stage in approx an hour or so.

**Comment by Eccenux** [comment]
Thanks for resolving this so fast! 👍 
I thought I would have to rewrite whole bunch of my extension.

**Comment by Marat-Tanalin** [comment]
It’s not enough just to resolve the specific issue.

This case should be a **lesson** for the linter authors that they should **double-check** the **quality** and **reasonability** of **each** new invented check **before** shipping it the way that prevents addon many developers from updating their addons on the ground of **pure theory** and **blind formality**.
